### PR TITLE
story #154807739: make clear if we encounter package that has partial or no validation support

### PR DESCRIPTION
**Pull request by Onur101010**

### CODE REVIEW

File: core/src/org/sbml/jsbml/validator/offline/constraints/SBMLDocumentConstraints.java [hunk 1]
```diff
@@ -47,6 +48,8 @@
    * 
    */
   public static final transient String XML_DECLARED_ENCODING = "jsbml.validator.xml.declared.encoding";
+  
+  private HashMap<String, Integer> packageValidationAvailability;
```
Comment (draeger) [inline core/src/org/sbml/jsbml/validator/offline/constraints/SBMLDocumentConstraints.java:54]: Even though it is private, I recommend to always declare variables with the least specific type, which is in this case `Map`.

**Comment by niko-rodrigue** [comment]
Have you tested it with several xml files to see if the proper error is thrown and if it is not thrown when it should not. You could create two folders that we will add into the set of validations files, name it like the others files with some fail and pass files

**Comment by Onur101010** [comment]
> t. You could create two folders that we will add into the set of validations files, name it like the others files with some fail and pass file

No not yet, okay I'll start and keep you updated.

**Comment by Onur101010** [comment]
Is it sufficient that the first activated package of the Document with no validation leads the constraint to be broken? Or do we want to have all packages to have no support to break the constraint 70001?

**Comment by Onur101010** [comment]
Okay perfect, then you can maybe doublecheck the validator with the new folders, it seems to work now.

**Comment by niko-rodrigue** [comment]
Yes, it is working but I would like, as  an exercise that you put the two
ValidationFunctions into one to see if you understood correctly how it is
working and to show later on an example of a ValidationFunction that handle
several validation errors at the same time.

**Comment by Onur101010** [comment]
Is something like this what you mean ?

**Comment by niko-rodrigue** [comment]
You are almost there. Here, you are re-using the same ValidationFunction twice, which is already good. What I want is to have only on ValidationFunction, called only once (meaning, you can remove the 'case CORE_70002:'). 
And then, in the implementation of the validationFunction, you are not allowed to test for the error code or to use CORE_70001 or CORE_70002 !

**Comment by niko-rodrigue** [comment]
Perfect, see how it make the method much simpler.

Last small details, please add a small comment(s) to explain that you are doing both errors in the same method. Also, you now can remove CORE_70002 from addErrorCodesForCheck.

**Pull request merged**